spec/models/jwt_api_entreprise_spec.rb

```diff
@@ -24,6 +31,88 @@
     it { is_expected.to have_and_belong_to_many(:roles) }
   end
 
+  describe '.access_request_survey_sent!' do
+    let!(:token) { create(:jwt_api_entreprise, access_request_survey_sent_trait) }
+
+    subject(:model) { described_class }
+
+    context 'when the access request survey was not sent' do
+      let(:access_request_survey_sent_trait) { :access_request_survey_not_sent }
+
+      it 'sets the flag to sent' do
+        expect {
+          model.access_request_survey_sent!(token.id)
+          token.reload
+        }.to change(token, :is_access_request_survey_sent?).from(false).to(true)
+      end
+
+      it { expect(model.access_request_survey_sent!(token.id)).to be_truthy }
+    end
+
+    context 'when the access request survey was sent' do
+      let(:access_request_survey_sent_trait) { :access_request_survey_sent }
+
+      it 'does not set the flag to sent' do
+        expect {
+          model.access_request_survey_sent!(token.id)
+          token.reload
+        }.not_to change(token, :is_access_request_survey_sent?)
+      end
+
+      it { expect(model.access_request_survey_sent!(token.id)).to be_falsey }
+    end
+  end
+
+  describe '.at_least_seven_days_ago_issued_tokens' do
+    let!(:token) { create(:jwt_api_entreprise, iat: iat_trait) }
+
+    subject(:model) { described_class }
```

Comment:
Le nombre de ligne est significatif, mais la sémantique l'est encore plus. Je vois que dans cet exemple, il y a encore une confusion entre le sujet du test et la valeur réelle qu'il retourne. J'en avais discuté ici : https://github.com/etalab/admin_api_entreprise/pull/92/#discussion_r604237078

Ceci étant, pour contraster chacun des trois tests **explicitement**, nous gagnerions à les contextualiser (dans un bloc `contexte`) et à les faire dériver en n'y modifiant qu'un seul paramètre, paramètre dont hérite le token. C'est une bonne manière de dire : dans tel contexte, j'ai tel comportement, tandis que dans tel autre contexte, j'ai plutôt tel autre comportement. C'est plus clair que de merger les caractéristiques différentes des contextes directement à l'intérieur d'un même contexte commun dans différentes variables.

Ça donne ce résultat là :

```ruby
subject { described_class }

let!(:token) { create(:jwt_api_entreprise, datetime_of_issue) }

context 'when the token was issued up to maximum 6 days ago' do
  let(:datetime_of_issue) { :less_than_seven_days_ago }

  its(:at_least_seven_days_ago_issued_tokens) { is_expected.not_to be_exist token.id }
end

context 'when the token was issued since at least 7 days ago' do
  let(:datetime_of_issue) { :about_seven_days_ago }

  its(:at_least_seven_days_ago_issued_tokens) { is_expected.to be_exist token.id }
end
```

Note : je propose de se limiter au minumum, soit 2 contextes (un 3ème serait IMO redondant).